Sources/CIImage+TestPattern.swift

```diff
@@ -0,0 +1,268 @@
+
+        // Put the label in the bottom left corner over the tile.
+        // Use a contrast color that is always readable, regardless of the tile's color.
+        let labelImage = CIImage.text(label, fontName: self.labelFont, fontSize:  size.height / 8.0, color: color.contrastOverlayColor)!
+        return labelImage.moved(to: CGPoint(x: 5, y: 5)).composited(over: tile)
+    }
+
+    /// Creates a swatch (line of colored tiles) for the given `color` for comparison.
+    /// Each tile will display the color in one of the `swatchColorSpaces`.
+    /// A label is added to the left next to the swatch displaying the color values.
+    private static func colorSwatch(for color: CIColor) -> CIImage {
+        // Generate a label to put next to the swatch.
+        let labelText = String(format: """
+            R: %.2f
+            G: %.2f
+            B: %.2f
+            """, color.red, color.green, color.blue)
+        var label = CIImage.text(labelText, fontName: ".AppleSystemUIFontMonospaced-Semibold", fontSize: self.tileSize.height / 7.0, color: .white, padding: 18)!
+
+        // Generate a color tile for each color space and place them next to each other.
+        var swatch = self.swatchColorSpaces.reduce(CIImage.empty()) { partialResult, entry in
+            var tile = CIImage.colorTile(for: color, colorSpace: entry.colorSpace, size: self.tileSize, label: entry.label)
+            tile = tile.translatedBy(dx: partialResult.extent.maxX.isInfinite ? 0 : partialResult.extent.maxX, dy: 0)
+            return tile.composited(over: partialResult)
+        }
+        // Also apply some round corners to the whole swatch.
+        swatch = swatch.withRoundedCorners(radius: self.tileCornerRadius)!
+
+        // Adjust placement of label and swatch.
+        label = label.centered(at: CGPoint(x: label.extent.midX, y: swatch.extent.midY))
+        swatch = swatch.moved(to: CGPoint(x: label.extent.maxX, y: 0))
+
+        return label.composited(over: swatch)
+    }
+
+    /// Creates a single tile (rectangle) filled with white (or gray) in the given `brightness` value with the given `size`.
+    /// The tile will be labeled with the `brightness` value and the corresponding nits value
+    /// (assuming standard brightness `1.0` = 500 nits).
+    private static func brightnessTile(for brightness: Double, size: CGSize) -> CIImage {
+        // Create the tile containing the brightness in all color channels.
+        var tile = CIImage.containing(value: brightness)!
+        tile = tile.cropped(to: CGRect(origin: .zero, size: size))
+
+        // Put a label in the bottom left corner over the tile that displays the brightness value (also in nits).
+        let labelText = String(format: "%.2f (%d nits)", brightness, Int(brightness * 500))
+        let labelColor = CIColor(extendedWhite: brightness)!.contrastOverlayColor
+        let label = CIImage.text(labelText, fontName: self.labelFont, fontSize: size.height / 8.0, color: labelColor)!
```

Comment:
If we want to be very safe we can add a `fontScaleFactor` since we divide by 8 multiple times.

Reply:
And one time `/ 7.0`. 😅
I didn't want to over-engineer this more than I already did. Otherwise, I'll end up writing a layout system for Core Image. 😬